src/extensions/default/InlineBoxModelEditor/BoxModelEditor.js

```diff
@@ -66,17 +66,37 @@ define(function(require, exports, module) {
         });
     }
 
-    PaddingValue.prototype.toString = function() {
+    BoxModelValue.prototype.toString = function() {
         return this.value + (this.value === 0 ? "" : this.unit);
     };
 
-    function PaddingEditor($parent, valueString, paddingChangeHandler) {
+    function BoxModelEditor($parent, valueString, BoxModelChangeHandler, type) {
         var self = this;
 
+        if(type === "margin"){
+          BoxModelTemplate = BoxModelTemplate.replace("{{SET_PADDING_FOR}}", "{{SET_MARGIN_FOR}}");
+        }
+        else{
+           BoxModelTemplate = BoxModelTemplate.replace("{{SET_MARGIN_FOR}}", "{{SET_PADDING_FOR}}");
+        }
+
         // Create the DOM structure, filling in localized strings via Mustache
-        self.$element = $(Mustache.render(PaddingTemplate, Strings));
+        self.$element = $(Mustache.render(BoxModelTemplate, Strings));
+        if(type === "margin"){
```

Comment:
let's not do this either. I would say pass the class name (just one) from the place where we send the type. So this function would take in the class-name and we would so something similar with the strings as we did above - 
```js
Strings["box-model-icon-class-name"] = iconClassName;
```

Then, in the template, change the class names to use a rendered value, for e.g., 
```html
<div class="right side-icon"></div>
```
will change to:
```html
<div class="right {{box-model-icon-class-name}}"></div>
```